content/200-concepts/100-components/02-prisma-client/260-client-extensions/120-query.mdx

```diff
@@ -147,6 +148,54 @@ const xprisma = prisma.$extends({
 })
 ```
 
+### Add or modify a custom query for a specific top-level client operation
+
+To apply a custom query to a specific top-level client operation:
```

Comment:
What does "apply a custom query" really mean here?
Are we overwriting the existing `$queryRaw` and making it do other things instead?
Or can we do something before the actual query is sent? Can we modify the raw query that is later executed?

Reply:
Yep, so I based this on previous paragraphs and assumed this was explained somewhere in the beginning or via prior examples. So yes, you can make it do different things, like wrap it in a transaction, append another query before it, or even build a cache in front of it. And yes, you can also modify the args before they are sent. So really like a middleware here.

Reply:
That definitely is clearer.

Does it apply only to raw queries though or all of them?
If only raw, we could mention that?
If not, why the xample only for raw ones?

Reply:
It's only the queries if I understood correctly because as the examples show you have to extend those within the `query` key of the `.$extends` object parameter.

I specified "top-level client _query_ operations" here:

> To apply custom behavior to a specific top-level client query operation:

Is that too subtle?

Reply:
query is also `findMany` - I am unclear if those also belong to "specific top-level client operation", and if so, why we only have raw queries as examples.

Reply:
Ah, that makes sense.
That I did not understand directly, might indicate that we could just be explicit and call these "raw queries" instead of " top-level client operation" if they are all raw query methods after all.

Reply:
Agreed, I'll make that change now 👍🏻

Reply:
Okay, I've made the change to the section title. It is now: 

> Modify top-level raw query operations:

The previous name after re-reading it a few times was also misleading as it seemed to suggest you could add your own custom queries to the top-level API which currently isn't possible. 

> Add or modify a custom query for top-level raw query operations:

Reply:
Sounds good to me. Can you confirm @millsp this is what it does?